<?xml version="1.0"?>
<launch>
  <include file="$(find lms1xx)/launch/LMS1xx.launch"/>

  <node pkg="platform_control" name="write_serial" type="write_serial" />

  <node pkg="laser_filters" type="scan_to_scan_filter_chain" output="screen" name="laser_filter">
	  <rosparam command="load" file="$(find pg_tracking)/launch/box_filter.yaml" />
  </node>
  

  <node pkg="pg_tracking" name="pg_tracking" type="tracking.py" output="screen">
	  <param name="constant_distance" value="3.5" />
	  <param name="KI" value="10.0" />
	  <param name="KP" value="100.0" />
	  
  </node>

  <node name="obstacle_extractor" pkg="obstacle_detector" type="obstacle_extractor_node">
	<remap from="scan" to="scan_filtered"/>
    <param name="active"               value="true"/>
    <param name="use_scan"             value="true"/>
    <param name="use_pcl"              value="false"/>

    <param name="use_split_and_merge"    value="true"/>
    <param name="circles_from_visibles"  value="true"/>
    <param name="discard_converted_segments" value="true"/>
    <param name="transform_coordinates"  value="true"/>

    <param name="min_group_points"     value="8"/>

    <param name="max_group_distance"   value="0.1"/>
    <param name="distance_proportion"  value="0.00628"/>
    <param name="max_split_distance"   value="0.2"/>
    <param name="max_merge_separation" value="1.5"/>
    <param name="max_merge_spread"     value="0.2"/>
    <param name="max_circle_radius"    value="0.1"/>
    <param name="radius_enlargement"   value="0.3"/>

    <param name="frame_id"             value="laser"/>
  </node>


</launch>
